### PR TITLE
Added toarray alias for totable

### PR DESCRIPTION
**Pull request by svermeulen**
I usually hear the word "table" as a reference to any lua tables generically rather than one that is meant to be used as a sequential set of data, so thought this would make sense as an alias

### CODE REVIEW

**Comment by igormunkin** [comment]
@svermeulen, thanks for your patch! I would like to share some thoughts regarding the naming.

**TL;DR:** `tosequence`/`toseq` aliases looks better (at least to me) if you need another one for `totable` (I doubt you do).

--- 

Strictly saying, there are no arrays in Lua, but I see your point regarding the naming: you would like to emphasis the fact there is no "holes" in numerical part of Lua table, hence the length value of the table is defined and correct. Unfortunately, such a structure is actually called "array" and even "array with/without holes" in [Lua 5.1 Reference manual](https://www.lua.org/manual/5.1/manual.html#2.5.5). However, starting from [Lua 5.2](https://www.lua.org/manual/5.2/manual.html#3.4.6) it is called a "sequence", that (IMHO) is way much better naming for this particular sort of Lua tables.

My reasoning follows this rule of thumb: array (as a continuous block of homogeneous items) is always considered with **its** length or capacity; sequence is a continuous number of elements with no terminator (i.e. `nil`) in the middle of it, that sounds closer to single-linked list (just consider the way [`table.maxn`](https://github.com/LuaJIT/LuaJIT/blob/v2.1/src/lib_table.c#L57-L77) is calculated in LuaJIT).

All in all, everything above are just my perception of the Lua world, so if somebody with the power of merge is OK with your naming, I can't insist on mine. "You must choose, but choose wisely" ;)

---

And of course, please add some tests for your alias and update the docs, otherwise it will be broken or unused in the future.

**Review by igormunkin — CHANGES_REQUESTED** [review]
@svermeulen, thanks for the fixes and sorry for stalling the discussion.

I still have a couple of nits:
* It's better to squash all the commits into the single one with a nice commit message.
* The test for `toseq` alias is still missing. I'd rather implement the test checking the fact that both `toseq` and `tosequence` are aliases for `totable` instead of writing functional tests for them.
* Docs still say not a word regarding the new aliases.

Otherwise, I have no objections and ask to proceed the second reviewer with the patch.

**Review by igormunkin — CHANGES_REQUESTED** [review]
@svermeulen, this PR hasn't been updated for over a year. Are you going to address the review comments above?
